test-utils/src/main/kotlin/dev/restate/e2e/utils/RestateDeployer.kt

```diff
@@ -183,25 +184,41 @@ private constructor(
       this.additionalContainers[entry.first] = entry.second
     }
 
-    fun withEnv(key: String, value: String) = apply { this.additionalEnv[key] = value }
+    fun withEnv(key: String, value: String) = apply { this.runtimeContainerEnvs[key] = value }
 
-    fun withEnv(map: Map<String, String>) = apply { this.additionalEnv.putAll(map) }
+    fun withEnv(map: Map<String, String>) = apply { this.runtimeContainerEnvs.putAll(map) }
 
     fun withInvokerRetryPolicy(policy: RetryPolicy) = apply { this.invokerRetryPolicy = policy }
 
     fun disableTracesExport() = apply { this.enableTracesExport = false }
 
     fun withConfig(configSchema: RestateConfigSchema) = apply { this.configSchema = configSchema }
 
-    fun build() =
-        RestateDeployer(
-            runtimeDeployments,
-            serviceEndpoints,
-            additionalContainers,
-            additionalEnv + (invokerRetryPolicy?.toInvokerSetupEnv() ?: emptyMap()),
-            runtimeContainer,
-            enableTracesExport,
-            configSchema)
+    fun disableLoadingRestateEnv() = apply { this.loadEnvs = false }
+
+    fun build(): RestateDeployer {
+      val loadedRuntimeContainerEnvs =
+          if (this.loadEnvs) {
+            System.getenv().filterKeys {
+              (it.uppercase().startsWith("RESTATE_") &&
+                  it.uppercase() != "RESTATE_RUNTIME_CONTAINER") ||
```

Comment:
While touching it, should we rename the env variable to `RESTATE_CONTAINER_IMAGE`?